src/include/binder/binder_context.h

```diff
@@ -14,6 +14,7 @@
 #include <string>
 
 #include "type/types.h"
+#include "concurrency/transaction.h"
```

Comment:
@haoxianghua Forward declaration means that you type the declaration of the class at the top of this .h file by yourself, and include the real declaration of the class in the .cpp file. This will significantly reduce coupling and compilation time of the code base. Just as how we declare TableRef here:
https://github.com/cmu-db/peloton/blob/master/src/include/binder/binder_context.h#L21

You can google "forward declaration c++" for more details.